### PR TITLE
Improve scrolling to new messages

### PR DESCRIPTION
**Pull request by csduarte**
Experimental scrolling implementation for scrolling to new messages

@jarredwitt

### CODE REVIEW

**Comment by enahum** [comment]
@hmhealey can you try and test this please

**Comment by jarredwitt** [comment]
@enahum added highlighted post case

File: app/components/post_list/post_list.js [hunk 1]
```diff
@@ -140,17 +171,33 @@ export default class PostList extends PureComponent {
...
+                    scrollToMessage: true
+                });
+            }
+        }
+    }
+
     renderItem = ({item, index}) => {
         if (item === START_OF_NEW_MESSAGES) {
             this.newMessagesIndex = index;
+            this.moreNewMessages = this.props.postIds.length === index + 2;
```
Comment (hmhealey) [inline app/components/post_list/post_list.js]: I'm not sure I understand how you get this value

**Comment by hmhealey** [comment]
There's something else that just occurred to me, how does this handle screen rotation, and could we leverage the measurement code to help with https://mattermost.atlassian.net/browse/ICU-448? The issue with that PR is that the list items are being measured with the wrong screen size which causes them to overlap on the screen

File: app/components/post_list/post_list.js (hunk 1 above)
Comment (jarredwitt) [reply app/components/post_list/post_list.js]: So we pull in 15 posts and a date header for the initial post display. If there is a new message `START_OF_NEW_MESSAGES` is added as well making our overall `postIds.length` 17. So if the new message index is 15 which would technically be the end of the post list, we have to add 2 to account for the extra 2 items that are added on.

File: app/components/post_list/new_messages_divider.js [hunk 2]
```diff
@@ -14,22 +14,37 @@ import {makeStyleSheetFromTheme} from 'app/utils/theme';
 function NewMessagesDivider(props) {
     const style = getStyleFromTheme(props.theme);
 
+    let Text = (
```
[the remark this answers is not in the document]
Comment (jarredwitt) [reply app/components/post_list/new_messages_divider.js]: Will do

File: app/components/post_list/index.js [hunk 3]
```diff
@@ -5,18 +5,25 @@ import {bindActionCreators} from 'redux';
...
+        const postIds = preparePostIds(state, ownProps);
+        const measureCellLayout = postIds.indexOf(START_OF_NEW_MESSAGES) > -1 || Boolean(ownProps.highlightPostId);
+
+        const {deviceHeight} = state.device.dimension;
+
         return {
-            postIds: preparePostIds(state, ownProps),
+            deviceHeight,
+            measureCellLayout,
+            initialBatchToRender: 15,
```
[the remark this answers is not in the document]
Comment (jarredwitt) [reply app/components/post_list/index.js]: Yeah that may be leftover code from some other experiments with this. I'll set it to a constant.

**Comment by jarredwitt** [comment]
@hmhealey changes made

**Comment by enahum** [comment]
This should not me merged until we create the release branch which should happen on monday

File: app/components/post_list/post_list.js (hunk 1 above)
Comment (hmhealey) [reply app/components/post_list/post_list.js]: Oh okay, I think I understand that. Could you add a comment explaining this so we can remember what this does when we come across this code in the future?

**Pull request merged**